Lib/test/test_types.py

```diff
@@ -756,6 +757,23 @@ def __module__(self):
         with self.assertRaises(ZeroDivisionError):
             str | TypeVar()
 
+    @cpython_only
+    def test_or_type_operator_reference_cycle(self):
+        if not hasattr(sys, 'gettotalrefcount'):
+            self.skipTest('Cannot get total reference count.')
+        gc.collect()
+        before = sys.gettotalrefcount()
+        for _ in range(30):
+            T = typing.TypeVar('T')
+            U = int | list[T]
+            T.blah = U
+            del T
+            del U
+        gc.collect()
+        leeway = 64
```

Comment:
Why is it so large? It means that leaking 2 references per iteration is still acceptable.

The threshold  should be less than the number of iterations. I tested, and 7 is enough. Maybe 10 just for the case.

Reply:
Without the fix, each iteration leaks 16 references. So 64 would still trigger after 4 iterations. Though you're right that it seems abit excessive. So I'll reduce it to 15, which won't allow for more than 1 reference leak per 2 iterations. Thanks.